File: Modules/Common/Techs/SIE.Web.Tech/Processs/Commands/ProcessPackingUnitCommand.js
SIE.defineCommand('SIE.Web.Tech.Processs.Commands.ProcessPackingUnitCommand', {
    extend: 'SIE.cmd.LookupCommandBase',
    userConfig: {
        dataParams: { specKeyPrototyName: 'PackageUnitId', targetClassName: 'SIE.Packages.Packages.PackingUnit' },
    },
    meta: { text: "选择", group: "edit", iconCls: "icon-PlaylistCheck icon-blue" },
    canExecute: function (view) {
        var p = view.getParent();
        if (!p) { return true; }
        var entity = p.getCurrent();
        if (entity === null) {
            return false;
        };
        return entity.data.CreateBy > 0 && (entity.data.Type === 20 || entity.data.Type === 40);
    },
    save: function (win) {
        var me = this;
        var indata = {};
        var selections = this._targetView.getSelection();
        if (selections && selections.length > 0) {
            var operationDatas = [];
            SIE.each(selections, function (item) {
                var packageUnitId = item.getId();
                if (me._sourceViewSelectItems.indexOf(packageUnitId) === -1) {
                    var PackageUnit = { ProcessId: me._sourceId, PackageUnitId: packageUnitId };
                    operationDatas.push(PackageUnit);
                }
            });
            indata = operationDatas;
            me._targetView.execute({
                data: indata,
                success: function (res) {
                    win.close();
                    me._ownerView.loadChildData(true);
                }
            }, me._ownerView);
        }
        else {
            SIE.Msg.showWarning('没有可提交的数据'.t());
        }
    },
});

File: Modules/Common/Techs/SIE.Web.Tech/Processs/Commands/ProcessPackingUnitCommand.cs
using SIE.Domain;
using SIE.Tech.Processs;
using SIE.Web.Command;
using System;
using System.Collections.Generic;

namespace SIE.Web.Tech.Processs.Commands
{
    /// <summary>
    /// 工序包装单位命令
    /// </summary>
    [JsCommand("SIE.Web.Tech.Processs.Commands.ProcessPackingUnitCommand")]
    class ProcessPackingUnitCommand : ViewCommand
    {
        /// <summary>
        /// 工序包装单位命令执行方法
        /// </summary>
        /// <param name="args">视图参数</param>
        /// <param name="scope">作用域</param>
        /// <returns>执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var meta = ClientEntities.Find(args.Type);
            var savedData = RF.Find(meta.EntityType).NewList();
            var defectList = args.Data.ToJsonObject<List<ProcessPackingUnit>>();
            Check.NotNullOrEmpty(defectList, nameof(defectList));
            if (defectList == null || defectList.Count == 0)
            {
                throw new ArgumentNullException("{0}数据参数不能为空".L10nFormat(nameof(defectList)));
            }

            foreach (var item in defectList)
            {
                var processPackingUnit = new ProcessPackingUnit();
                processPackingUnit.PackageUnitId = item.PackageUnitId;
                processPackingUnit.ProcessId = item.ProcessId;
                savedData.Add(processPackingUnit);
            }

            RF.Save(savedData);
            return true;
        }
    }
}